src/js/helpers/nodesHelper.spec.js

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { knotxNodes } from './nodesHelper';
+
+beforeEach(() => {
+    document.body.innerHTML = '';
+})
+
+test('Empty DOM results in empty fragment array', () => {
+    const fragments = knotxNodes()
+    expect(fragments).toStrictEqual([])
+});
+
+test('DOM without fragments results in empty array', () => {
+    document.body.innerHTML = 
+        '<div>' +
+        '   <span>Test</span>' +
+        '</div>';
+
+
+    const fragments = knotxNodes();
+    expect(fragments).toStrictEqual([]);
+});
+
+test('Fragment without data script tag is ignored', () => {
+    document.body.innerHTML = 
+        '<span>Test</span>' +
+        '<!-- data-knotx-id="test" -->' +
+        '<div>Test fragment</div>' +
+        '<!-- data-knotx-id="test" -->';
+
+    const fragments = knotxNodes();
+    expect(fragments).toStrictEqual([]);
+});
+
+test('Fragment with empty data is parsed', () => {
+    document.body.innerHTML = constructMarkupWithFragment('test', '{}', '<div>Test fragment</div>');
+
+    const fragments = knotxNodes();
+    expect(fragments).toHaveLength(1);
+});
+
+test('Empty fragment is parsed', () => {
+    document.body.innerHTML = constructMarkupWithFragment('test', '{}', '');
+
+    const fragments = knotxNodes();
+    expect(fragments).toHaveLength(1);
+});
+
+test('Fragment without debug json throws syntax error', () => {
+    document.body.innerHTML = constructMarkupWithFragment('test', '', '<div>Test fragment</div>')
+
+    const execution = () => knotxNodes();
+    expect(execution).toThrow(SyntaxError);
+});
+
+test('Fragment with incorrect json throws syntax error', () => {

```

Comment:
Is this desired behaviour to throw syntax error?

Reply:
Those unit tests are just documentation of the current behavior. What is should behave like is open for discussion, especially since it has poor error handling.